engine/include/cubos/engine/collisions/broad_phase_collisions.hpp

```diff
@@ -0,0 +1,103 @@
+/// @file
+/// @brief Contains the class for the BroadPhaseCollisions resource.
+
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+
+#include <glm/glm.hpp>
+
+#include <cubos/core/ecs/entity_manager.hpp>
+
+using cubos::core::ecs::Entity;
+
+namespace cubos::engine
+{
+    /// Used to store data used in broad phase collision detection.
+    class BroadPhaseCollisions final
+    {
+    public:
+        using Candidate = std::pair<Entity, Entity>;
+
+        BroadPhaseCollisions() = default;
+        ~BroadPhaseCollisions() = default;
+
+        /// Collision type for each pair of colliders.
+        enum class CollisionType
+        {
+            BoxBox = 0,
+            BoxCapsule,
+            CapsuleCapsule,
+            BoxPlane,
+            CapsulePlane,
+            PlanePlane,
+            BoxSimplex,
+            CapsuleSimplex,
+            PlaneSimplex,
+            SimplexSimplex,
+
+            AMOUNT ///< Used to get the amount of collision types.
+        };
+
+        /// Updates and sorts the markers for sweep and prune.
+        /// @param position Function used to get the bounds of the entity.
+        template <typename F>
+        void updateMarkers(F position);
```

Comment:
Since this is a template function, you must define it in the header file.
Overall, don't you think it would be easier to understand whats going on if we moved those methods to systems?
Also, document what F does.

Reply:
I commented this out, the function will be on the system